pkg/controller/controllerutil/controllerutil.go

```diff
@@ -32,6 +34,86 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
+var (
+	// Will store memoized controller namespace
+	inClusterNamespace     string
+	inClusterNamespacePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
```

Comment:
Should this be a const?

Reply:
Normally yes, but am overriding the value for testing the function, and the alternatives weren't much better.